mediator/med_map_mod.F90

```diff
@@ -929,33 +931,46 @@ subroutine med_map_field_packed(FBSrc, FBDst, FBFracSrc, field_normOne, packed_d
-    type(ESMF_FieldBundle)    , intent(inout) :: FBDst
-    type(ESMF_Field)          , intent(in)    :: field_normOne(:)  ! array over mapping types
-    type(ESMF_FieldBundle)    , intent(in)    :: FBFracSrc         ! fraction field bundle for source
-    type(packed_data_type)    , intent(inout) :: packed_data(:)    ! array over mapping types
-    type(ESMF_RouteHandle)    , intent(inout) :: routehandles(:)
-    integer                   , intent(out)   :: rc
+    type(ESMF_FieldBundle)          , intent(in)    :: FBSrc
+    type(ESMF_FieldBundle)          , intent(inout) :: FBDst
+    type(ESMF_Field)                , intent(in)    :: field_normOne(:)  ! array over mapping types
+    type(ESMF_FieldBundle)          , intent(in)    :: FBFracSrc         ! fraction field bundle for source
+    type(packed_data_type)          , intent(inout) :: packed_data(:)    ! array over mapping types
+    type(ESMF_RouteHandle)          , intent(inout) :: routehandles(:)
+    type(ESMF_FieldBundle), optional, intent(in)    :: FBDat             ! data field bundle 
+    logical, optional               , intent(in)    :: use_data          ! skip mapping and use data instead
+    integer, optional               , intent(out)   :: rc
 
     ! local variables
-    integer                    :: nf, nu, np, n
-    integer                    :: fieldcount
-    integer                    :: mapindex
-    integer                    :: ungriddedUBound(1)
-    real(r8), pointer          :: dataptr1d(:)
-    real(r8), pointer          :: dataptr2d(:,:)
-    real(r8), pointer          :: dataptr2d_packed(:,:)
-    type(ESMF_Field)           :: field_fracsrc
-    type(ESMF_Field), pointer  :: fieldlist_src(:)
-    type(ESMF_Field), pointer  :: fieldlist_dst(:)
-    real(r8), pointer          :: data_norm(:)
-    real(r8), pointer          :: data_dst(:,:)
-    character(len=*), parameter  :: subname=' (med_map_mod:med_map_field_packed) '
+    integer                       :: nf, nu, np, n, nfd
+    integer                       :: fieldcount, fieldcount_dat
+    integer                       :: mapindex
+    integer                       :: ungriddedUBound(1)
+    real(r8), pointer             :: dataptr(:)
+    real(r8), pointer             :: dataptr1d(:)
+    real(r8), pointer             :: dataptr2d(:,:)
+    real(r8), pointer             :: dataptr2d_packed(:,:)
+    type(ESMF_Field)              :: field_fracsrc
+    type(ESMF_Field), pointer     :: fieldlist_src(:)
+    type(ESMF_Field), pointer     :: fieldlist_dst(:)
+    type(ESMF_Field), pointer     :: fieldlist_dat(:)
+    real(r8), pointer             :: data_norm(:)
+    real(r8), pointer             :: data_dst(:,:)
+    character(cl)                 :: field_name
+    character(cl), allocatable    :: field_namelist_dat(:)
+    logical                       :: skip_mapping
+    type(ESMF_Region_Flag)        :: zeroregion
+    real(ESMF_KIND_R8), parameter :: fillValue = 9.99e20_ESMF_KIND_R8
```

Comment:
This is a fill value used previously. So, I did not add it but just move to one level up. When @danrosen25 implemented it for the HAFS, he used this and it is also used in FV3 and also HYCOM. So, it would be hard to change it at this point but I think with CDEPS inline capability, UFS will not need it in the near future. Not sure.

Reply:
If this same value is used in other locations as you imply here then there should be a single parameter used in all of those locations, it should not be implemented several times as you appear to be doing.

Reply:
I mean it is used in other model component. So, at this point this is the value that is choked by the components to understand unmapped regions and fill them in the component side. This would be useless in the near future when all HAFS applications switches to CDEPS inline implementation. Then, if we want we could change this value and make it same with `SHR_CONST_SPVAL` but before that changing this will break the exiting HAFS tests.